### PR TITLE
MRG: Fix scaled determination in `fastgather` and `fastmultigather`

### PR DESCRIPTION
**Pull request by ctb**
Fixes https://github.com/sourmash-bio/sourmash_plugin_branchwater/issues/491

This PR updates `fastgather` and `fastmultigather` against rocksdb to pick scaled automatically, rather than using a default value of 1000. `fastmultigather` now exits when no search sketches are found.

`scaled=1000` is removed as the default throughout.

It also eliminates the use of usize throughout the codebase in favor of u32/u64, following https://github.com/sourmash-bio/sourmash/pull/3364.

TODO:
- [x] test fail exit on fastmultigather with no matching foo
- [x] fix scaled selection for all the other commands, too
- [x] finish the usize job started in https://github.com/sourmash-bio/sourmash_plugin_branchwater/pull/500

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review & merge @bluegenes !

File: src/fastgather.rs [hunk 1]
```diff
@@ -12,8 +13,7 @@ use crate::utils::{
 pub fn fastgather(
     query_filepath: String,
     against_filepath: String,
-    threshold_bp: usize,
-    scaled: usize,
```
Comment (bluegenes) [inline src/fastgather.rs:16]: wait, so now it's not possible to pass in a different scaled -- the collection _must_ be at the scaled you want to search with?

Comment (bluegenes) [reply src/fastgather.rs:16]: oh nm, it's in selection, right?

**Review by bluegenes — APPROVED** [review]
lgtm

File: src/fastgather.rs (hunk 1 above)
Comment (ctb) [reply src/fastgather.rs:16]: yep!

**Pull request merged**